File: solutions/project-euler/003/003.py
#!/usr/bin/env python3

from math import sqrt

LIMIT = 600851475143


def is_prime(num):
    if num % 2 == 0:
        return False
    for current in range(3, int(sqrt(num)) + 1):
        if num % current == 0:
            return False
    return True


for i in range(1, int(sqrt(LIMIT))):
    if LIMIT % i == 0 and is_prime(i):
        prime = i

print(prime)

File: solutions/project-euler/003/003.c
#include <math.h>
#include <stdio.h>

int16_t is_prime(int32_t n) {
    int32_t i;

    if (n == 2) {
        return 1;
    }
    if (n % 2 == 0 || n < 2) {
        return 0;
    }

    for (i = 3; i <= sqrt(n); i += 2) {
        if (n % i == 0) {
            return 0;
        }
    }
    return 1;
}

int16_t main() {
    int16_t i;
    int32_t n = 600851475143;
    int32_t r;

    for (i = 2; i <= sqrt(n); i++) {
        if (n % i == 0 && is_prime(i)) {
            r = i;
        }
    }
    printf("%lld\n", r);

    return 0;
}
